### PR TITLE
Add docs for Nomad autodiscover and add_nomad_metadata

### PR DESCRIPTION
**Pull request by jsoriano**
Add docs for `add_nomad_metadata`.

Documentation for autodiscover was already there, but was not being
included in the docs build.

Docs are focused on Filebeat, and for autodiscover only included there. Current features are more useful with logs, they are not aware of network information yet.

Related to https://github.com/elastic/beats/issues/16853.

### CODE REVIEW

File: libbeat/docs/shared-autodiscover.asciidoc [hunk 1]
```diff
@@ -493,7 +493,7 @@ ifdef::autodiscoverAWSEC2[]
 [float]
 ===== Amazon EC2s
 
-*Note: This provider is experimental*
+experimental[]
```
Comment (kaiyan-sheng) [inline libbeat/docs/shared-autodiscover.asciidoc:496]: Thank you for fixing this!!

**Comment by jsoriano** [comment]
jenkins run the tests please

**Pull request merged**